hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -232,17 +250,18 @@ void emitCommitMetrics(String instantTime, HoodieCommitMetadata metadata, String
    * Main API to run bootstrap to hudi.
    */
   public void bootstrap(Option<Map<String, String>> extraMetadata) {
-    if (rollbackPending) {
-      rollBackInflightBootstrap();
-    }
+    // TODO : MULTIWRITER -> check if failed bootstrap files can be cleaned later
     HoodieTable<T, I, K, O> table = getTableAndInitCtx(WriteOperationType.UPSERT, HoodieTimeline.METADATA_BOOTSTRAP_INSTANT_TS);
+    if (isEager(config.getFailedWritesCleanPolicy())) {
```

Comment:
The `isEager` is needed for multiple operations, not just `rollbackFailedBootstrap()`, changed it to do check within method but cannot do operations inside eagerCleanFailedWrites. This will need more refactoring (if possible), I can open a follow up ticket and take it up right after the release to adhere against release timeline.